### PR TITLE
Hacker role

### PR DESCRIPTION
**Pull request by Gorott**

### CODE REVIEW

File: commands/game/hack.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            let guy2 = message.guild.members.cache.find((m) => m.nickname === args[1])
+            if (!message.member.roles.cache.has(alive.id)) return message.channel.send("You are dead")
+            if (isNight != "yes") return message.channel.send("You can only hack when everyone it's night")
+            if (args.length < 1 || args.length > 2) return message.channel.send("You have either chosen more than 2 or 0 players.")
+            let hack = []
+            for (let i = 0; i < args.length; i++) {
+                if (i == 0) {
+                    if (!guy1 || guy1.id == message.author.id) return message.reply("Invalid Target!")
+                    if (!guy1.roles.cache.has(alive.id)) return message.channel.send("You cannot hack a dead player!")
+                    check.push(guy1.nickname)
```
Comment (thewilloftheshadow) [inline commands/game/hack.js:19]: pushing to the wrong variable

Comment (thewilloftheshadow) [inline commands/game/hack.js:19]: pushing to the wrong variable

File: commands/narrator/day.js [hunk 2]
```diff
@@ -2982,6 +2983,271 @@ module.exports = {
...
         }, 4500)
 
+        //hacker
+        let THEHACKER = []
+        for (let z = 0; z < hacker.length; z++) {
+            let chan1 = message.guild.channnels.cache.get(hacker[z])
+            for (let y = 1; y < 17; y++) {
+                let tempguy = message.guild.members.cache.find((m) => m.nickname === y.toString())
+                if (tempguy) {
+                    THEHACKER = tempguy
```
Comment (thewilloftheshadow) [inline commands/narrator/day.js:2993]: what is this line even doing? you're changing the array that was there into a GuildMember and then never using it?

**Review by thewilloftheshadow — CHANGES_REQUESTED** [review]
Need to add lines to +reset to delete the hack_ and hacked_ variables

File: commands/narrator/day.js (hunk 2 above)
Comment (Gorott) [reply commands/narrator/day.js:2993]: Basically the array is an mistake
But this whole thing is for the ss thing where it checks if the guy is in the array thekiller however i put the thing after ss check so its pretty pointless

File: commands/game/hack.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+                if (i == 0) {
+                    if (!guy1 || guy1.id == message.author.id) return message.reply("Invalid Target!")
+                    if (!guy1.roles.cache.has(alive.id)) return message.channel.send("You cannot hack a dead player!")
+                    hack.push(guy1.nickname)
+                } else {
+                    if (!guy2 || guy2.id == message.author.id) return message.reply("Invalid Target!")
+                    if (!guy2.roles.cache.has(alive.id)) return message.channel.send("You cannot hack a dead player!")
+                    hack.push(guy2.nickname)
+                }
+                db.set(`hack_${message.channel.id}`, hack)
```
Comment (thewilloftheshadow) [inline commands/game/hack.js:25]: reset this variable too

**Review by thewilloftheshadow — CHANGES_REQUESTED** [review]

**Review by thewilloftheshadow — APPROVED** [review]

**Pull request merged**